### PR TITLE
Ce 1622 insights notification tracking

### PR DESCRIPTION
**Pull request by kamilkoterba**
https://wikia-inc.atlassian.net/browse/CE-1622

Added onShow callback handler for BannerNotification
Added tracking for clicks on BannerNotification links within Insights loop

@Wikia/community-engineering

### CODE REVIEW

**Comment by bkoval** [comment]
LGTM :+1:

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	function ($, tracker, Querystring) {
+		'use strict';
+
+		var qs = new Querystring(),
+			insightType = qs.getVal('insights', null),
+			itemStatus = qs.getVal('item_status', null);
+
+		/* If there's no itemStatus use action parameter */
+		if (itemStatus === null) {
+			itemStatus = qs.getVal('action', null);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: Why you get 'action' param here?

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+				label: insightType+'-'+itemStatus+'-'+linkType.data
+			};
+			tracker.track(trackingParams);
+		}
+
+		function init(event, bannerNotification) {
+			/* Setup click events within BannerNotification */
+			bannerNotification.$element.find('#InsightsBackToListButton').click('back-to-list', onClickTrack);
+			bannerNotification.$element.find('#InsightsNextPageButton').click('next-page', onClickTrack);
+			bannerNotification.$element.find('button.close').click('dismiss', onClickTrack);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: I will consider using 'keydow' event to send tracking request earlier

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * checks for item_status url param to log action status (fixed/notfixed)
+ */
+define('ext.wikia.Insights.LoopNotificationTracking',
+	['jquery', 'wikia.tracker', 'wikia.querystring'],
+	function ($, tracker, Querystring) {
+		'use strict';
+
+		var qs = new Querystring(),
+			insightType = qs.getVal('insights', null),
+			itemStatus = qs.getVal('item_status', null);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: I know part with VE is not finished now but we need to consider how to pass this param when VE is enabled

File: extensions/wikia/Insights/scripts/LoopNotification.js [hunk 4]
```diff
@@ -21,9 +28,12 @@ require(['jquery', 'BannerNotification', 'wikia.querystring', 'wikia.window'],
 	showNotification = function(html) {
 		if (html) {
 			var msgType = getNotificationType(),
-				$parent = getParent();
+				$parent = getParent(),
+				bn;
 
-			new BannerNotification(html, msgType, $parent).show();
+			bn = new BannerNotification(html, msgType, $parent);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/LoopNotification.js]: Can we create this object when we declare ```bn``` var?

**Comment by kamilkoterba** [comment]
Comments will be addressed within https://github.com/Wikia/app/pull/7024

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js (hunk 1 above)
Comment (kamilkoterba) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: That's for identifying edit mode in logs - I'll change it to use notificationType that you prepared recently here https://github.com/Wikia/app/pull/7037/files#diff-274f0ab383736c0231df0070ba374d63R34

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js (hunk 2 above)
Comment (kamilkoterba) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: added mousedown and keydown for keycode 13

File: extensions/wikia/Insights/scripts/LoopNotification.js (hunk 4 above)
Comment (kamilkoterba) [inline extensions/wikia/Insights/scripts/LoopNotification.js]: ok

**Pull request merged**